TeachingRecordSystem/tests/TeachingRecordSystem.SupportUi.Tests/PageTests/SupportTasks/ApiTrnRequests/Resolve/MatchesTests.cs

```diff
@@ -452,13 +452,97 @@ public async Task Post_CreateNewRecordChosen_UpdatesStateAndRedirects()
+            $"/support-tasks/api-trn-requests/{supportTask.SupportTaskReference}/matches?{journeyInstance.GetUniqueIdQueryParameter()}")
+        {
+            Content = new FormUrlEncodedContentBuilder() { { "PersonId", selectedPersonId } }
+        };
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        journeyInstance = await ReloadJourneyInstance(journeyInstance);
+        Assert.Null(journeyInstance.State.FirstNameSource);
+        Assert.Null(journeyInstance.State.MiddleNameSource);
+        Assert.Null(journeyInstance.State.LastNameSource);
+        Assert.Null(journeyInstance.State.DateOfBirthSource);
+        Assert.Null(journeyInstance.State.EmailAddressSource);
+        Assert.Null(journeyInstance.State.NationalInsuranceNumberSource);
+    }
+
+    [Fact]
+    public async Task Post_PersonIdNotChangedFromState_DoesNotClearPersonAttributes()
+    {
+        // Arrange
+        var applicationUser = await TestData.CreateApplicationUserAsync();
+        var supportTask = await TestData.CreateApiTrnRequestSupportTaskAsync(applicationUser.UserId);
+
+        var selectedPersonId = ResolveApiTrnRequestState.CreateNewRecordPersonIdSentinel;
+
+        var journeyInstance = await CreateJourneyInstance(
+            supportTask.SupportTaskReference,
+            new ResolveApiTrnRequestState
+            {
+                PersonId = selectedPersonId,
+                FirstNameSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                MiddleNameSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                LastNameSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                DateOfBirthSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                EmailAddressSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord,
+                NationalInsuranceNumberSource = ResolveApiTrnRequestState.PersonAttributeSource.ExistingRecord
+            });
+
+        var request = new HttpRequestMessage(
+            HttpMethod.Post,
+            $"/support-tasks/api-trn-requests/{supportTask.SupportTaskReference}/matches?{journeyInstance.GetUniqueIdQueryParameter()}")
+        {
+            Content = new FormUrlEncodedContentBuilder() { { "PersonId", selectedPersonId } }
+        };
+
+        // Act
+        var response = await HttpClient.SendAsync(request);
+
+        // Assert
+        journeyInstance = await ReloadJourneyInstance(journeyInstance);
+        Assert.NotNull(journeyInstance.State.FirstNameSource);
+        Assert.NotNull(journeyInstance.State.MiddleNameSource);
+        Assert.NotNull(journeyInstance.State.LastNameSource);
+        Assert.NotNull(journeyInstance.State.DateOfBirthSource);
+        Assert.NotNull(journeyInstance.State.EmailAddressSource);
+        Assert.NotNull(journeyInstance.State.NationalInsuranceNumberSource);
```

Comment:
Same here, what does it mean that they are not null?

Reply:
Inverse of above, so if a user goes from Matches to Merge then CheckAnswers (so the *Source properties are populated) then back to Matches but does not change the Person then the Source attributes are not reset, so the user doesn't have to go through Merge again.